src/polkadot/PolkadotTools.ts

```diff
@@ -72,8 +81,10 @@ export class PolkadotTools implements EdgeCurrencyTools {
         [`${pluginId}Key`]: Buffer.from(secretKey).toString('hex')
       }
     } else if (isHex(userInput)) {
+      const hex = userInput.replace(/^0x/i, '')
+      if (hex.length !== 64 && hex.length !== 128) throw new Error('InvalidPrivateKey')
```

Comment:
The userInput needs to be tested using the library. this length check is not sufficient to ensure the string is valid